endpoints/v2/test/test_v2auth.py

```diff
@@ -217,3 +217,33 @@ def test_generate_registry_jwt(
 
     assert decoded["access"] == expected_access
     assert len(decoded["context"][CLAIM_TUF_ROOTS]) == len(expected_scopes)
+
+
+@pytest.mark.parametrize(
+    "push_private, visibility",
+    [
+        (
+            True,
+            "private",
+        ),
+        (
+            False,
+            "public",
+        ),
+    ],
+)
+def test_generate_registry_jwt_visibility(push_private, visibility, app, client):
+    scope = "repository:devtable/newrepo:pull,push,*"
+    username = "devtable"
+    password = "password"
+    expected_code = 200
+    expected_scopes = ["devtable/newrepo:push,pull,*"]
+    try:
+        original_app.config["CREATE_PRIVATE_REPO_ON_PUSH"] = push_private
+        test_generate_registry_jwt(
```

Comment:
While you _can_ call another test, I would say it's generally preferable to decouple it so that it's easier to debug in the future.

Reply:
I considered it but chose this route as the clearest. I want to run the test only isolating a config change.

Reply:
The issue is I can't really tell from looking at the test what it's doing. If you're just calling another test, you could just parametrize the existing one with your constraints.

Reply:
@kleesc Moved to the existing test